src/kOS/Binding/MissionSettings.cs

```diff
@@ -84,32 +84,10 @@ public override void AddTo(SharedObjects shared)
 
         public object VesselShortcutGetter(string name)
         {
-            ISuffixResult suffix =ship.GetSuffix(name);
+            ISuffixResult suffix = ship.GetSuffix(name);
             if (!suffix.HasValue)
                 suffix.Invoke(sharedObj.Cpu);
             return suffix.Value;
         }
-
-        public override void Update()
-        {
-            base.Update();
-            if (ship == null)
-            {
-                ship = VesselTarget.CreateOrGetExisting(sharedObj);
-                ship.LinkCount++;
-            }
-            else if (ship.Vessel == null)
-            {
-                ship.LinkCount--;
-                ship = VesselTarget.CreateOrGetExisting(sharedObj);
-                ship.LinkCount++;
-            }
-            else if (!ship.Vessel.id.Equals(sharedObj.Vessel.id))
```

Comment:
This probably handled docking (when the vessel becomes part of station).
Keeping the link for few updates after query by `:SHIP` could still be viable (to prevent re-creation of part-values),
but I would rather move such logic to shared objects (some `KeepVesselTargetAlive()` that would reset some counter which would be changed and checked every tick to remove the reference if it reaches 0 / max).